File: 1541_잃어버린 괄호/첫번째 시도_성공.cpp
#include <iostream>
#include<sstream>
#include<string>
#include <stack>

using namespace std;
int main() {
	stringstream ss;
	string input;
	stack<int> number;

	int num;
	int sum = 0;
	cin >> input;
	ss.str(input);
	bool check = false;
	while (ss >> num) {
		if (num > 0) {
			if (sum == 0) {
				number.push(num);
			}
			else {
				if (!check) {
					number.push(num);
				}
				else {
					sum += num;
				}
			}
		}
		else {
			if (!check) {
				check = true;
				sum += num * (-1);
			}
			else {
				number.push((-1)*(sum));
				sum = num * (-1);
				check = true;
			}	
		}
	}
	if(check)
		number.push((-1)*(sum));
	int output = 0;
	for (int i = 0; !number.empty(); i++) {
		output += number.top();
		number.pop();
	}
	cout << output;
	return 0;
}

File: Backjoon/14916_거스름 돈/첫번째 시도_성공.py
input_num = int(input())
check = []
for i in range(1,input_num+1):
    if (i%2 != 0) &(i %5 != 0) and i < 5:
        check.append(-1)
    else:
        if (i % 5 == 0):
            check.append(i//5)
        elif (i <10 and i % 2 == 0):
                check.append(i//2)
        else :
            check.append(check[i-6] + 1)
        
print(check[input_num-1])
        
